docs/node/timeouts.md

```diff
@@ -0,0 +1,93 @@
+
+Servers can interact with this timeout via the [handler context](https://connectrpc.com/docs/node/implementing-services#context).
+Depending on your needs, there are a few ways to approach it:
+
+The first way is via an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) 
+on the context. Using this signal, route handlers can then tell if the timeout specified 
+by the client was reached and abort their processes accordingly. The `AbortSignal` can be found
+via the property name `signal`.
+
+The signal can be passed to other functions or used to gracefully stop processes when the timeout is reached.
+Using `signal` works for any operation you might want to call as long as the API supports it. 
+
+```ts
+import type { HandlerContext } from "@bufbuild/connect";
+
+const say = async (req: SayRequest, ctx: HandlerContext) => {
+
+    ctx.signal.aborted; // true if timed out
+    ctx.signal.reason; // an error with code deadline_exceeded if timed out
+
+    // raises an error with code deadline_exceeded if timed out
+    ctx.signal.throwIfAborted();
+
+    // the AbortSignal can be passed to other functions
+    await longRunning(ctx.signal);
+
+    return new SayResponse({sentence: `You said: ${req.sentence}`});
+};
+```
+
+A second way to interact with the timeout value is via the `timeoutMs()` function 
+on the handler context. If the current request has a timeout, this function 
+returns the remaining time.
+
+Using the `timeoutMs()` function is preferable when invoking upstream RPC calls 
+because it is more efficient and robust - you have a guarantee that the peer is 
+aware of the deadline, regardless of network issues.
+
+```ts
+import type { HandlerContext } from "@bufbuild/connect";
+
+const say = async (req: SayRequest, ctx: HandlerContext) => {
+  
+    // If a timeout was set on the call to this service, the timeoutMs() method 
+    // returns the remaining time in milliseconds. 
+
+    // Passing the value to an upstream client call propagates the timeout.
+    await upstreamClient.someCall({}, { timeoutMs: ctx.timeoutMs() });
+
+    return new SayResponse({sentence: `You said: ${req.sentence}`});
+};
+```
+
+## `ConnectRouter` options
+
+In addition, to server-side support for timeouts, there are two related options on `ConnectRouter`
+that help working with timeouts:
+
+- `maxTimeoutMs?: number`<br/>
+  The maximum value for timeouts that clients may specify.
+  If a client requests a timeout that is greater than `maxTimeoutMs`,
+  the server responds with the error code `invalid_argument`.
+- `shutdownSignal?: AbortSignal`<br/>
+  If this signal is aborted, all deadline-related `AbortSignals` in handler contexts
+  will be aborted as well. This gives implementations a chance to wrap up
+  work before the server process is killed and shut down servers gracefully.
+
+An example of setting these values when using the [Fastify plugin](http://localhost:3001/docs/node/server-plugins#fastify):
```

Comment:
Debated whether we needed to show this. I think users might be a bit confused on how to set connect router options because we don't really have them documented anywhere.